### PR TITLE
Update docs bucket URL

### PR DESCRIPTION
**Pull request by Robert-Steiner**
### Description

- Updated docs bucket URL to `flower.ai`.

### Checklist

- [ ] Implement proposed change
- [ ] Write tests
- [ ] Update [documentation](https://flower.dev/docs/writing-documentation.html)
- [ ] Update the changelog entry below
- [ ] Make CI checks pass
- [ ] Ping maintainers on [Slack](https://flower.dev/join-slack/) (channel `#contributions`)

### Changelog entry

<skip>

### CODE REVIEW

File: dev/deploy-docs.sh [hunk 1]
```diff
@@ -16,27 +16,34 @@
...
 cd build/html
-aws s3 sync --delete --exclude ".*" --exclude "v/*" --cache-control "no-cache" ./ s3://flower.dev/docs/examples
+aws s3 sync --delete --exclude ".*" --exclude "v/*" --cache-control "no-cache" ./ s3://flower.ai/docs/examples
+
+# Build and deploy Flower Datasets docs
+cd "$ROOT"
+cd datasets/doc
+make docs
+cd build/html
+aws s3 sync --delete --exclude ".*" --exclude "v/*" --cache-control "no-cache" ./ s3://flower.ai/docs/datasets
```
Comment (tanertopal) [inline dev/deploy-docs.sh:49]: @Robert-Steiner @charlesbvll This is new, right? How have we deployed datasets docs until now?

Comment (Robert-Steiner) [reply dev/deploy-docs.sh:49]: @tanertopal @charlesbvll Do we want to keep it or should i remove it?

Comment (tanertopal) [reply dev/deploy-docs.sh:49]: @Robert-Steiner @charlesbvll 
Given we are going for more automation/security I am ok with deleting the script at this point.

Comment (Robert-Steiner) [reply dev/deploy-docs.sh]: done

**Review by tanertopal — APPROVED** [review]

**Pull request merged**